### ISSUE TITLE
Virus in archive ?

### ISSUE THREAD

**Issue by orenault** [SeleniumHQ/selenium#2062]
Hi, I try to download:

http://selenium.googlecode.com/files/selenium-server-2.37.0.zip

It is blocked by my enterprise antivirus:

Content contained "PUS:NetTool.Win32.ProxySwitcher.d" virus. Details: Virus: PUS:NetTool.Win32.ProxySwitcher.d; File: selenium-server-2.37.0.zip; Sub File: Memory region/selenium-2.37.0/selenium-server-2.37.0.jar; Vendor: Kaspersky Labs; Engine error code: 0x00010000; Engine version: 8.1.8.79; Pattern version: 160506.054000.7590664; Pattern date: 2016.05.06 05:40:00

Can you check if there's really an issue with viruses in this archive (and the others) or if it's an issue on the antivirus server ?

Thanks
Olivier

**Comment by ddavison** [comment]
our official download repository is: https://selenium-release.storage.googleapis.com/index.html

i'm not sure if it was ever that url you provided.. could someone else chime in?

**Comment by lukeis** [comment]
those were the old urls for download from google code... but we have no control over them anymore and google will likely kill them at some point in the future.

The file in question though is valid and doesn't contain any viruses, although I think there used to be issues in our old releases where virus scanners would flag selenium due to some of the executables that were bundled inside the jars (which aren't actually viruses).

We can't do anything with these old releases... if there is a problem with a current release in the storage url provided by Daniel, please log a new issue.

**Comment by pmneve** [comment]
My corporate resource (who has permission to download from https://selenium-release.storage.googleapis.com/index.html)  is get the same virus warning on selenium-server-standalone.jar 2.52 and 2.53 which we desperately need for dealing with IE 11 and other new releases of browsers.

Can this be re-opened or should I log a new issue?

**Comment by lukeis** [comment]
actually this appears to be an old issue with your anti-virus software -

https://github.com/seleniumhq/selenium-google-code-issue-archive/issues/789

we do have code in the selenium project that allows one to manipulate the system proxy on windows (it's a feature, not a virus :) ). Only thing you can do is to try to get an exception to the rule from whomever admin's it at your company.

**Comment by pmneve** [comment]
Luke,

Thanks for your quick reply.
That certainly looks like the problem.
We have not tried to get the selenium-server-standalone.jar downloading inside the firewall before (it has been bundled with another application that doesn't seem to be checked by the same av...)

pat

**Comment by cgoldberg** [comment]
some history and background info:

Selenium users have complained about the standalone server getting flagged by security software for years.

The false positive is triggered by `hudsuckr.exe` inside `selenium-server-standalone-*.jar`. Several virus scanners identify it as malware.

so WTF is `hudsuckr.exe`?
 
Hudsuckr is a proxy configuration tool for Windows that alters proxy settings inside the Windows registry.  It was developed by Thoughtworks, and the source code was integrated with the old Selenium-RC codebase.  Oddly, when RC was migrated to Selenium 2, the compiled binary (from 2007) was comitted directly to the repo (without source code).  Source can still be viewed in various other code archives on the interwebs.  It is still used by RC, and gets executed from `WindowsProxyManager.java` (https://github.com/SeleniumHQ/selenium/blob/master/java/server/src/org/openqa/selenium/server/browserlaunchers/WindowsProxyManager.java) 

 * hudsuckr source code (C++): http://code.openhub.net/file?fid=P7bzq9FrBC_Yz45twxRnu33iJhU&cid=vZia_FrIJ1E&fp=26531
 * blog post introducing hudsuckr: http://darkforge.blogspot.com/2007/10/hudsuckr-advanced-windows-proxy.html

On a good note, RC is getting dropped in Selenium 3, so there will be no need to ship this shady proxy switcher anymore.
  
TBH. I'd consider it malware also :)

**Comment by pmneve** [comment]
Corey,

This doesn't help us download the jar.

Is there a way we can get a standalone jar that doesn't include RC at all?

We really cannot wait for Selenium 3 to get on with critical testing in IE.

pat

**Comment by cgoldberg** [comment]
@pat,
see comments below:

> This doesn't help us download the jar. 

I know... It was not my intention to help you download the file.  I was providing information about what causes the issue (since it comes up so often).  But seriously, why can't you hop on a computer that's not protected by your corporate virus scanner, download the jar, and sneakernet it onto your test system?

> Is there a way we can get a standalone jar that doesn't include RC at all?

Not from the selenium maintainers...  But you can clone the repo, prune the RC code, and build it yourself (It's open source afterall).  You might need to tweak some build configuration, but it's pretty trivial to do your own custom build.

-Corey

**Comment by pmneve** [comment]
Corey,

As for sneakernet, it is pretty well locked down here.  No discs, no thumb drives, etc. allowed (even recognized) on the corporate workstations/laptops.

As for building it my self I am not a java or c++ jock and would have a learning curve I can't afford right now, so that's not practical at the moment.
But I'll see if there is someone available who can do it.

And will definately pursue getting corporate security to let that 'PUS' get through.

Thanks much for your help!!

pat
